src/core/lib/iomgr/endpoint.cc

```diff
@@ -61,3 +61,10 @@ int grpc_endpoint_get_fd(grpc_endpoint* ep) { return ep->vtable->get_fd(ep); }
 grpc_resource_user* grpc_endpoint_get_resource_user(grpc_endpoint* ep) {
   return ep->vtable->get_resource_user(ep);
 }
+
+bool grpc_endpoint_can_track_err(grpc_endpoint* ep) {
+  if (ep->vtable->can_track_err != nullptr) {
```

Comment:
No need for this check if you change this field to a bool

Reply:
I think you should still remove this check. Just ensure that every endpoint has an implementation, even if that impl is just return false